lib/src/client.dart

```diff
@@ -795,27 +795,49 @@ class Client extends MatrixApi {
         ));
       }
     }
+    try {
+      // Start a new direct chat
+      final roomId = await createRoom(
+        invite: [mxid],
+        isDirect: true,
+        preset: preset,
+        initialState: initialState,
+        powerLevelContentOverride: powerLevelContentOverride,
+      );
 
-    // Start a new direct chat
-    final roomId = await createRoom(
-      invite: [mxid],
-      isDirect: true,
-      preset: preset,
-      initialState: initialState,
-      powerLevelContentOverride: powerLevelContentOverride,
-    );
-
-    if (waitForSync) {
-      final room = getRoomById(roomId);
-      if (room == null || room.membership != Membership.join) {
-        // Wait for room actually appears in sync
-        await waitForRoomInSync(roomId, join: true);
+      if (waitForSync) {
+        final room = getRoomById(roomId);
+        if (room == null || room.membership != Membership.join) {
+          // Wait for room actually appears in sync
+          await waitForRoomInSync(roomId, join: true);
+        }
       }
-    }
 
-    await Room(id: roomId, client: this).addToDirectChat(mxid);
+      await Room(id: roomId, client: this).addToDirectChat(mxid);
 
-    return roomId;
+      return roomId;
+    } catch (e) {
+      /// This looks up the room id of the room that was just created.
+      /// The room is empty since createRoom threw an error. To make sure not to
+      /// delete other empty rooms by accident, the creation time is checked.
+      /// Should be fixed in Synapse and then be removed
```

Comment:
You should really document, why this throws an error, because the synapse code tries to handle all errors ahead of time, so it is not clear, why you are getting an error on room creation. If the problem is invites, we can for example move those out of the room creation.